src/msix/PAL/Signature/OpenSSL/SignatureValidator.cpp

```diff
@@ -131,7 +131,7 @@ namespace MSIX
                             M_ASN1_OCTET_STRING_print(extbio.get(), ext->value);
                         }
                         // null terminate the string.
-                        BIO_write(extbio.get(), "", 1);
+                        BIO_write(extbio.get(), "\0", 1);
```

Comment:
There is no functional difference with your change; an empty string literal is still a char[1] containing a null byte.  You have made a char[2] containing two null bytes.

The function should be returning 1 if it is working correctly (before or after the change).  If it already was, then I don't think that this is the source of the issue.

The better code here is different anyway:
1. Don't write the null byte
2. Construct the string from `bptr->data` and `bptr->length` to avoid the unnecessary strlen call (`std::string_view` would be even better but if I remember correctly this code is C++14)

Reply:
That makes sense. Making that change now.